### DISCUSSION TITLE
Occasional redirect to api/auth/error

### DISCUSSION THREAD

**Discussion by mschewe** [nextauthjs/next-auth#6012] [Help]
I noticed recently some occasional redirects to the URL `api/auth/error`. There is no further information provided in the URL. No logs in the console and on the server (debugging in next-auth activated). 

I found someone else with this behavior. See #5983 

> It works if the NextAuth options passed to unstable_getServerSession are spreaded into a new object.
This issue is not specific to version 4.18.0, I tested it with 4.17.0 and 4.12.3 and had the similar issue.

I now spread the `authOptions` into a new object. But I am not sure if this fixes the problem.
It somehow feels like it works a few hours after a new deployment, but then it fails. Deleting cookies and cache does not fix this problem. 

It happened on version `4.13.0` and `4.18.0` with Next `12.2.5`. 
The application is deployed to Vercel. 

I really wonder what is happening there. Can someone explain it to me?

**Comment by 0xfulgore** [comment]
So I have fixed this for me; I deploy on render.com and rely on their environment variables for NEXTAUTH_URL. I have since changed this behaviour and my start command now is no longer just "yarn start" but "NEXTAUTH_URL=xxx yarn start"

This has resolved the issue for me. I am unsure as to why in the other instances the env variable seems to get lost / confused.
